src/core/utils/Histogram.hpp

```diff
@@ -0,0 +1,20 @@
+namespace Utils {
+
+/**
+* \brief Returns the unravelled index of the provided flat index.
+*        Therefore is the inversion of flattening an ndims dimensional index.
+* @param len_dims an int array of length ndims containing the lengths of the dimensions. (Input)
+* @param ndims int denoting the number of dimensions. (Input)
+* @flattened_index an int denoting the flat index. (Input)
+* @unravelled_index_out an int array with length ndims where the unflat indices are written to. (Output)
+*/
+inline void unravel_index(const int* const len_dims, const int ndims, const int flattened_index, int* unravelled_index_out){
+	//idea taken from http://codinghighway.com/2014/02/22/c-multi-dimensional-arrays-part-2-flattened-to-unflattened-index/
+	int mul[ndims];
```

Comment:
I'm sure this is some compiler extension.  Please use `std::vector`.

Reply:
I wont touch that function since it's used in the reaction ensemble (where I actually ripped it off).

Reply:
That's a variable-length array, which is not legal syntax in C++. It is a GNU extension and supported by Clang, Intel and possibly others.
You can safely replace this line with `std::vector<int> mul(ndims);` to get the exact same functionality.